migrateBuckets/migrateBuckets.js

```diff
@@ -213,88 +222,90 @@ function _changeOneObjectAccount(bucket, object, log, cb) {
     const { Key, VersionId } = object;
+    keyInProgress = Key;
+    versionIdInProgress = VersionId;
     const decodedId = VersionId ? versionIdUtils.decode(VersionId) : undefined;
 
     // if versiondId is undefined, retrieval will return most current object md
     const getOptions = { versionId: decodedId };
 
     mdclient.getObject(bucketName, Key, log.getSerializedUids(),
     (err, data) => {
-        const objectMD = JSON.parse(data);
         if (err) {
-            log.error(`Error getting object metadata: ${err}`);
+            ++nErrors;
+            log.error(`Error getting object ${Key} metadata: ${err}`);
             return cb(err);
         }
+        const objectMD = JSON.parse(data);
         const objectId = objectMD['owner-id'];
         const objectOwner = objectMD['owner-display-name'];
         if (objectId === destCanonicalId && objectOwner === destOwnerName) {
-            log.debug(`Object ${Key} id already updated`);
+            log.info(`Object ${Key} id already updated`);
             return cb();
         }
         objectMD['owner-id'] = destCanonicalId;
         objectMD['owner-display-name'] = destOwnerName;
         const putOptions = {};
         if (VersionId) {
-            putOptions.versionId = VersionId;
+            putOptions.versionId = decodedId;
         }
         return mdclient.putObject(bucketName, Key, JSON.stringify(objectMD),
         log.getSerializedUids(), err => {
             if (err) {
+                ++nErrors;
                 log.error(`Error updating object ${Key} metadata: ${err}`);
                 return cb(err);
             }
-            log.debug(`Object ${Key} id successfully updated`);
+            ++nObjProcessed;
+            log.info(`Object ${Key}, Version ID ${VersionId} is ` +
+                'successfully updated');
             return cb();
         }, putOptions);
     }, getOptions);
 }
 
 function _changeObjectsAccount(bucket, log, cb) {
     const bucketName = bucket.bucketName;
-    const versioningEnabled = bucket.versioningEnabled;
-    const listMethod = versioningEnabled ? 'listObjectVersions' : 'listObjects';
-    const dataKey = versioningEnabled ? 'Versions' : 'Contents';
     let moreObjects = true;
     let objectsArr = [];
     let marker = null;
     async.whilst(
-        () => moreObjects === true,
-        done => {
+        () => moreObjects, done => {
             const listParams = { Bucket: bucketName };
             if (marker) {
                 listParams.Marker = marker;
             }
-            s3[listMethod](listParams, (err, data) => {
+            // if versioning is not enabled on bucket, VersionId will be null
+            s3.listObjectVersions(listParams, (err, data) => {
                 if (err) {
+                    ++nErrors;
                     return done(err);
                 }
-                objectsArr = objectsArr.concat(data[dataKey]);
+                objectsArr = objectsArr.concat(data.Versions);
```

Comment:
Maybe you forgot to remove `objectsArr` after the change to process versions during listing?